config/routes.rb

```diff
@@ -0,0 +1,32 @@
+Rails.application.routes.draw do
+  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  # root 'main#root'
+  get '/', to: 'main#index', as: 'root'
+
+  resources :merchants, only: [:show]
+  get '/merchants/:merchant_id/products', to: 'merchants#products', as: 'merchant_products'
+
+  resources :products, except: :destroy do
+    resources :reviews, only: [:new, :create]
+  end
+
+  get 'categories/:id/products', to: 'products#index', as: 'products_categories'
+
+  # Add resource not resources because a customer will only have one shopping cart and doesn't create an index route
+  resource :cart, only: [:show]
+
+  resources :payments, only: [:show]
+
+  resources :orders, only: [:show, :index] do
+    resources :payments, only: [:index, :new, :create]
+  end
```

Comment:
I'm not sure why the `payments` routes are nested under `orders` here. Seems to me that the routes for a payments ought not to be parameterized at all, but instead take cart info from the `session`.